File: gpt4o_simple_test.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
Простой тест GPT-4o Audio Preview для транскрипции аудио
"""

import os
import requests
from pathlib import Path
from dotenv import load_dotenv

# Загрузка переменных окружения
load_dotenv()

# Получаем API ключ
api_key = os.getenv("OPENAI_API_KEY")
if not api_key:
    print("❌ OPENAI_API_KEY не найден в переменных окружения")
    exit(1)

# Путь к аудиофайлу
audio_file_path = "server/uploads/dbbdcdc0-3e44-49d8-96c4-07afe0f3943d.wav"

# API URL
api_url = "https://api.openai.com/v1/audio/transcriptions"

print(f"Тестирование транскрипции файла: {audio_file_path}")

# Открываем файл и отправляем запрос
try:
    with open(audio_file_path, "rb") as audio_file:
        print("Отправка запроса на транскрипцию...")
        
        headers = {
            "Authorization": f"Bearer {api_key}"
        }
        
        files = {
            "file": audio_file
        }
        
        data = {
            "model": "whisper-1"
        }
        
        response = requests.post(
            api_url,
            headers=headers,
            files=files,
            data=data
        )
        
        if response.status_code == 200:
            result = response.json()
            print("\nРезультат транскрипции:")
            print("-------------------------------------------")
            print(result["text"])
            print("-------------------------------------------")
            print("✅ Тест успешно завершен!")
        else:
            print(f"❌ Ошибка ({response.status_code}):")
            print(response.text)
            
except Exception as e:
    print(f"❌ Ошибка при выполнении запроса: {e}")

File: gpt4o_simple_test.js
/**
 * Простой тест GPT-4o аудио транскрипции
 */
import 'dotenv/config';
import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import FormData from 'form-data';
import fetch from 'node-fetch';

const __filename = fileURLToPath(import.meta.url);
const __dirname = path.dirname(__filename);

// Путь к тестовому аудиофайлу
const sampleAudioPath = path.join(__dirname, 'server', 'uploads', 'dbbdcdc0-3e44-49d8-96c4-07afe0f3943d.wav');

async function transcribeAudio() {
  try {
    console.log(`Тестирование транскрипции файла: ${sampleAudioPath}`);
    
    if (!fs.existsSync(sampleAudioPath)) {
      console.error(`Файл не найден: ${sampleAudioPath}`);
      return;
    }
    
    const formData = new FormData();
    formData.append('file', fs.createReadStream(sampleAudioPath));
    formData.append('model', 'whisper-1');
    
    console.log('Отправка запроса на транскрипцию...');
    const response = await fetch('https://api.openai.com/v1/audio/transcriptions', {
      method: 'POST',
      headers: {
        'Authorization': `Bearer ${process.env.OPENAI_API_KEY}`
      },
      body: formData
    });
    
    if (response.ok) {
      const result = await response.json();
      console.log('\nРезультат транскрипции:');
      console.log('-------------------------------------------');
      console.log(result.text);
      console.log('-------------------------------------------');
      console.log('✅ Тест успешно завершен!');
    } else {
      const error = await response.text();
      console.error(`Ошибка (${response.status}):`, error);
    }
  } catch (error) {
    console.error('Ошибка при выполнении теста:', error.message);
  }
}

// Выполняем тест
transcribeAudio();